### PR TITLE
docs: ADR for support of Gaia-X Participant Self-Description in IdentityHub

### PR DESCRIPTION
**Pull request by bscholtes1A**
## What this PR changes/adds

Provides ADR for IdentityHub extension enabling of Gaia-X Participant Self-Description (creation and exposure).

## Why it does that

Documentation.

## Linked Issue(s)

Contributes to #4.

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] assigned appropriate label? (exclude from changelog with label `no-changelog`)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [Etiquette for pull requests](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/pr_etiquette.md) for details_)

### CODE REVIEW

File: docs/developer/decision-records/2023-01-03-gaiax-participant-self-description/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+## Decision
+
+Provide an extension for the [IdentityHub](https://github.com/eclipse-edc/IdentityHub) component enabling creation, signature and
+exposure of Gaia-X participant self-description.
+
+## Rationale
+
+In a Gaia-X dataspace, each participant must describe themselves using standardised, machine comprehensible metadata called self-description (SD).
+A SD is essentially a [W3C Verifiable Credential (VC)](https://www.w3.org/TR/vc-data-model/) composed of a set of standardised claims describing the participant (e.g. name, LEI number, headquarter localisation...)
```
Comment (clange) [inline docs/developer/decision-records/2023-01-03-gaiax-participant-self-description/README.md:11]:
```suggestion
According to the specification of the [Gaia-X credential format](https://gaia-x.gitlab.io/technical-committee/federation-services/icam/credential_format/), a SD is essentially a [W3C Verifiable Presentation (VP)](https://www.w3.org/TR/vc-data-model/), comprising one or more Verifiable Credentials (VC), each composed of a set of standardised claims describing the Gaia-X entity (e.g., in the case of a Participant, its name, LEI number, headquarter localisation, …).
```

File: docs/developer/decision-records/2023-01-03-gaiax-participant-self-description/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+## Decision
+
+Provide an extension for the [IdentityHub](https://github.com/eclipse-edc/IdentityHub) component enabling creation, signature and
+exposure of Gaia-X participant self-description.
+
+## Rationale
+
+In a Gaia-X dataspace, each participant must describe themselves using standardised, machine comprehensible metadata called self-description (SD).
+A SD is essentially a [W3C Verifiable Credential (VC)](https://www.w3.org/TR/vc-data-model/) composed of a set of standardised claims describing the participant (e.g. name, LEI number, headquarter localisation...) 
+and signed by an authority endorsed by Gaia-X called a Trust Anchor.
```
Comment (clange) [inline docs/developer/decision-records/2023-01-03-gaiax-participant-self-description/README.md:12]:
```suggestion
and signed by an authority endorsed by Gaia-X called a Trust Anchor (cf. the [Gaia-X Trust Framework](https://gaia-x.gitlab.io/policy-rules-committee/trust-framework/trust_anchors/)).
```

File: docs/developer/decision-records/2023-01-03-gaiax-participant-self-description/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+A SD is essentially a [W3C Verifiable Credential (VC)](https://www.w3.org/TR/vc-data-model/) composed of a set of standardised claims describing the participant (e.g. name, LEI number, headquarter localisation...) 
+and signed by an authority endorsed by Gaia-X called a Trust Anchor.  
+
+## Approach
+
+As described [here](https://gitlab.com/gaia-x/gaia-x-community/gaia-x-catalogue/catalogue-document/-/blob/85c23d34a6b1a37ab5982ec511493ce3ac8d54c6/01_self-descriptions.md), 
+the signature process can either be performed directly by the participant using an identity (certificate) provided by a Trust Anchor, or by the Trust Anchor itself. Both flavours
+should be supported by the extension.
+
+After the VC has been created, it is sent to the [Gaia-X Compliance service](https://compliance.gaia-x.eu/docs/#/Participant/ParticipantController_signContent) which validates
```
Comment (clange) [inline docs/developer/decision-records/2023-01-03-gaiax-participant-self-description/README.md:20]:
```suggestion
After the Self-Description has been created, it is sent to the [Gaia-X Compliance service](https://compliance.gaia-x.eu/docs/#/Participant/ParticipantController_signContent) which validates 
```

File: docs/developer/decision-records/2023-01-03-gaiax-participant-self-description/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+## Approach
+
+As described [here](https://gitlab.com/gaia-x/gaia-x-community/gaia-x-catalogue/catalogue-document/-/blob/85c23d34a6b1a37ab5982ec511493ce3ac8d54c6/01_self-descriptions.md), 
+the signature process can either be performed directly by the participant using an identity (certificate) provided by a Trust Anchor, or by the Trust Anchor itself. Both flavours
+should be supported by the extension.
+
+After the VC has been created, it is sent to the [Gaia-X Compliance service](https://compliance.gaia-x.eu/docs/#/Participant/ParticipantController_signContent) which validates 
+the format of the VC claims and asserts that is has been signed by an endorsed Trust Anchor. If successful, the API returns a compliance credential, which is another VC containing a hash of the first one.
+
+Both VCs are then persisted into a store. When a participant requests another to present its SD, these two VCs are assemble internally into a unique document returned to the caller.
```
Comment (clange) [inline docs/developer/decision-records/2023-01-03-gaiax-participant-self-description/README.md:23]: I wouldn't say "_both_ VCs", as the original Self-Description of a Participant may comprise more than just one VC.

When we store credentials locally, we do not necessarily need to wrap them into a VP.

However, the "unique document into which two/multiple VCs are assembled", isn't that actually a VP (comprising all VCs of the original SD=VP, plus the compliance credential)?

**Review by clange — COMMENTED** [review]
Improved compliance with Gaia-X specifications

File: docs/developer/decision-records/2023-01-03-gaiax-participant-self-description/README.md (hunk 4 above)
Comment (bscholtes1A) [reply docs/developer/decision-records/2023-01-03-gaiax-participant-self-description/README.md:23]: Hello @clange and thanks for the detailed review :)

Looking at the signing endpoint exposed by the [Gaia-X Compliance Services](https://compliance.gaia-x.eu/docs/#/Common/CommonController_signSelfDescription), it seems that they expect a Verifiable Credential as input, and not a VP. 

Then in the end if the signing process succeeds there will be actually 2 VC in the SD document (the one provided to the API + the compliance credential generated by the Gaia-X compliance service). 
And even if this SD document contains actually 2 VCs, it cannot be considered as a VP such as described in the [W3C spec](https://www.w3.org/TR/vc-data-model/#example-a-simple-example-of-a-verifiable-presentation), as here this document will look like:

```json
{
  "selfDescriptionCredential": {
    ...
  },
  "complianceCredential": {
    ...
  }
}
```

**Comment by bscholtes1A** [comment]
Hello @clange 
I updated the PR based on your comments. Could you please have another look?

Thanks in advance,
Benjamin.

**Review by jimmarino — APPROVED** [review]

**Review by wolf4ood — APPROVED** [review]

**Review by PeterKoen-MSFT — APPROVED** [review]

**Pull request merged**